component/remote/git/git.go

```diff
@@ -0,0 +1,287 @@
+// Package git implements the remote.git component.
+package git
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/remote/git/internal/vcs"
+	"github.com/grafana/agent/pkg/flow/logging/level"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "remote.git",
+		Args:    Arguments{},
+		Exports: Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the remote.git component.
+type Arguments struct {
+	Repository    string            `river:"repository,attr"`
+	Revision      string            `river:"revision,attr,optional"`
+	Path          string            `river:"path,attr"`
+	PullFrequency time.Duration     `river:"pull_frequency,attr,optional"`
+	GitAuthConfig vcs.GitAuthConfig `river:",squash"`
+}
+
+// Exports holds settings exported by remote.git.
+type Exports struct {
+	Content string `river:"content,attr"`
```

Comment:
Should this also be an OptionalSecret like in remote.http?
It's not good to keep secrets in git, but it's not unheard of... and better if we don't expose it in the UI? 🤔

Reply:
I would want it to be documented in `remote.git`, just the way that it's implemented in [remote.http](https://grafana.com/docs/agent/next/flow/reference/components/remote.http/). I don't think it would impact `module.git` docs.